src/components/select/select.spec.js

```diff
@@ -1486,6 +1506,29 @@ describe('<md-select>', function() {
     });
   });
 
+  // Only test with custom components on AngularJS 1.5+
+  if (angular.version.major === 1 && angular.version.minor >= 5) {
+    describe('with custom components', function() {
+
+      if (angular.version.major === 1 && angular.version.minor >= 7) {
+        // The $flushPendingTasks API is new in AngularJS 1.7 and is required to reproduce the issue
+        // in this test.
+        it('should re-validate when the options and required value change',
+          inject(function($flushPendingTasks) {
+            var nodes = $compile('<form name="testForm">' +
+              ' <required-form-field-component ng-ref="comp"></required-form-field-component>' +
+              '</form>')($rootScope);
+
+            $rootScope.$digest();
+            $flushPendingTasks();
+
+            expect($rootScope.comp.value).toBe('test');
+            expect($rootScope.testForm.value.$error.required).not.toBe(true);
```

Comment:
It is generally safer to assert the actual value: `expect(...).toBe(undefined)` (or whatever the value should be).
(Because for example, it might not be true, but be an object (as happens on forms for example), etc.)